datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -461,6 +467,67 @@ where
                     .map_err(|e| e.into_inner().unwrap_or(EthereumContractCallError::Timeout))
             })
     }
+
+    /// Request blocks by hash through JSON-RPC.
+    fn load_blocks_rpc(
+        &self,
+        logger: Logger,
+        ids: Vec<H256>,
+    ) -> impl Stream<Item = LightEthereumBlock, Error = Error> + Send {
+        let web3 = self.web3.clone();
+
+        stream::iter_ok::<_, Error>(ids.into_iter().map(move |hash| {
+            let web3 = web3.clone();
+            retry(format!("load block {}", hash), &logger)
+                .no_limit()
+                .timeout_secs(60)
+                .run(move || {
+                    web3.eth()
+                        .block_with_txs(BlockId::Hash(hash))
+                        .from_err::<Error>()
+                        .map_err(|e| e.compat())
+                        .and_then(move |block| {
+                            block.ok_or_else(|| {
+                                format_err!("eth node did not find block {:?}", hash).compat()
+                            })
+                        })
+                })
+                .from_err()
+        }))
+        .buffered(*BLOCK_BATCH_SIZE)
+    }
+
+    /// Request blocks ptrs for numbers through JSON-RPC.
+    ///
+    /// Reorg safety: If ids are numbers, they must be a final blocks.
+    fn load_block_ptrs_rpc(
+        &self,
+        logger: Logger,
+        block_nums: Vec<u64>,
+    ) -> impl Stream<Item = EthereumBlockPointer, Error = Error> + Send {
+        let web3 = self.web3.clone();
+
+        stream::iter_ok::<_, Error>(block_nums.into_iter().map(move |block_num| {
+            let web3 = web3.clone();
+            retry(format!("load block ptr {}", block_num), &logger)
+                .no_limit()
+                .timeout_secs(60)
+                .run(move || {
+                    web3.eth()
+                        .block(BlockId::Number(BlockNumber::Number(block_num)))
+                        .from_err::<Error>()
+                        .map_err(|e| e.compat())
+                        .and_then(move |block| {
+                            block.ok_or_else(|| {
+                                format_err!("eth node did not find block {:?}", block_num).compat()
```

Comment:
Same here. Having said this, if we already use `eth node` in all other errors in this module, you can just ignore me and resolve my comments.

Reply:
Looks like `eth node` is something I made up, `Ethereum node` is more consistent with the rest of the code base.